tests/unittests/test_request_timeout.py

```diff
@@ -0,0 +1,387 @@
+import unittest
+from unittest import mock
+
+from requests.exceptions import Timeout
+import tap_bing_ads
+from urllib.error import URLError
+from tap_bing_ads import CustomServiceClient
+import datetime
+    
+class MockClient():
+    '''Mocked ServiceClient class and it's method to pass the test case'''
+    def __init__(self, error):
+        self.error = error
+        self.count = 0
+
+    def GetCampaignsByAccountId(self, AccountId, CampaignType):
```

Comment:
Please add small comment for each function

Reply:
Added comments